### PR TITLE
Cleanup pushdown terminology

### LINKED ISSUE #3688: Cleanup pushdown terminology
### What should be done?

Our terminology for pushdowns is a mess. Since we don't have good terms, our code is a bit messy and has suspicious places that confuse different types of pushdowns: https://github.com/FerretDB/FerretDB/blob/22421e7e4e33a492169a65a495edc0d40b9afd8a/integration/query_test.go#L1005-L1012

We should clarify terms in our glossary, documentation, tests, code, and configuration (in that order).

The proposal:
* do not use "query pushdown" or just "pushdown" at all;
* use "filter pushdown" and "sort pushdown" for both inputs (flags, parameters) and outputs (explain results);
* make it clear that "sort pushdown" input also includes and implies `limit` and `skip` pushdowns;
* use "limit pushdown" for outputs only;
* separate "unsafe" from both "filter pushdown" and "sort pushdown" – there are already safe (capped collections) and unsafe sort pushdowns, and we might introduce unsafe filter pushdowns in the future.

### Where?

- [ ] glossary
- [x] the rest of the documentation
- [x] integration tests
- [x] unit tests
- [x] code
- [x] configuration flags

### Definition of Done

- handler updated;
- all backends updated;
- unit tests updated;
- integration/compatibility tests updated;
- spot refactorings done;
- user documentation updated.

### PR DESCRIPTION
**Pull request by noisersup**
# Description

Closes #3688.

## Readiness checklist

- [x] I added/updated unit tests (and they pass).
- [x] I added/updated integration/compatibility tests (and they pass).
- [x] I added/updated comments and checked rendering.
- [x] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: cmd/ferretdb/main.go [hunk 1]
```diff
@@ -87,6 +87,7 @@ var cli struct {
 		RecordsDir string `default:"" help:"Testing: directory for record files."`
 
 		DisableFilterPushdown    bool `default:"false" help:"Experimental: disable filter pushdown."`
+		EnableSortPushdown       bool `default:"false" help:"Experimental: enable sort pushdown."`
```
Comment (AlekSi) [inline cmd/ferretdb/main.go:90]: Why safe sort pushdown is not enabled by default, and why enabling it is considered experimental?

File: website/docs/pushdown.md [hunk 2]
```diff
@@ -17,6 +17,18 @@ To make this process more efficient, we minimize the amount of incoming data, by
 You can learn more about query pushdown in our [blog post](https://blog.ferretdb.io/ferretdb-fetches-data-query-pushdown/).
 :::
 
+## Types of pushdown
+
+Currently we support 3 different types of pushdown.
+
+The **Filter pushdown**, which is enabled by default, sends supported filters directly to the underlying database.
+It can be disabled with `--disable-filter-pushdown`.
+The **Sort Pushdown**, which can be enabled with `--enable-sort-pushdown`, pushes down sorting, `limit`, and `skip` methods.
```
Comment (rumyantseva) [inline website/docs/pushdown.md:26]: Hmm, what I don't understand is the difference between "safe" and "unsafe" sort pushdown... If we have a "safe" sort pushdown, why don't we enable it by default?

File: cmd/ferretdb/main.go (hunk 1 above)
Comment (AlekSi) [inline cmd/ferretdb/main.go:90]:
```suggestion
```

File: website/docs/pushdown.md (hunk 2 above)
Comment (noisersup) [reply website/docs/pushdown.md]: No longer part of this PR

File: integration/query_command_compat_test.go [hunk 3]
```diff
@@ -104,7 +104,7 @@ func testQueryCommandCompat(t *testing.T, testCases map[string]queryCommandCompa
 					var msg string
 					if setup.FilterPushdownDisabled() {
 						tc.resultPushdown = noPushdown
```
Comment (AlekSi) [inline integration/query_command_compat_test.go:106]: Please rename the field too

File: integration/query_test.go [hunk 4]
```diff
@@ -834,117 +834,117 @@ func TestQueryCommandUnsafeLimitPushDown(t *testing.T) {
...
-		len                 int                 // expected length of results
-		queryPushdown       resultPushdown      // optional, defaults to noPushdown
-		unsafeLimitPushdown bool                // optional, set true for expected pushdown for limit
-		err                 *mongo.CommandError // optional, expected error from MongoDB
-		altMessage          string              // optional, alternative error message for FerretDB, ignored if empty
-		skip                string              // optional, skip test with a specified reason
-		failsForSQLite      string              // optional, if set, the case is expected to fail for SQLite due to given issue
+		len            int                 // expected length of results
+		filterPushdown resultPushdown      // optional, defaults to noPushdown
+		limitPushdown  bool                // optional, set true for expected pushdown for limit
```
Comment (AlekSi) [inline integration/query_test.go:839]: Why is that `bool` and not `resultPushdown`?

File: website/docs/reference/glossary.md [hunk 5]
```diff
@@ -146,6 +146,12 @@ See [Operation modes](../configuration/operation-modes.md) for more details.
 An open source relational database.
 FerretDB uses PostgreSQL as a database engine.
 
+#### Pushdown
+
+The method of optimizing a query by reducing the amount of data read and processed in FerretDB, by moving some logic
+to underlying backend.
+See [Query pushdown](../pushdown.md) for more details.
```
Comment (AlekSi) [inline website/docs/reference/glossary.md:153]: Don't use "query pushdown". Do use "filter pushdown" and "sort pushdown"

File: website/docs/reference/glossary.md [hunk 6]
```diff
@@ -146,6 +146,12 @@ See [Operation modes](../configuration/operation-modes.md) for more details.
 An open source relational database.
 FerretDB uses PostgreSQL as a database engine.
 
+#### Pushdown
```
Comment (AlekSi) [inline website/docs/reference/glossary.md:149]: The issue is about clarifying terminology. We don't have a term "pushdown"

File: internal/handlers/sqlite/msg_explain.go [hunk 7]
```diff
@@ -137,9 +137,9 @@ func (h *Handler) MsgExplain(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg,
 
 			// our extensions
 			// TODO https://github.com/FerretDB/FerretDB/issues/3235
-			"pushdown", res.QueryPushdown,
-			"sortingPushdown", res.UnsafeSortPushdown,
-			"limitPushdown", res.UnsafeLimitPushdown,
+			"pushdown", res.FilterPushdown,
+			"sortingPushdown", res.SortPushdown,
```
Comment (AlekSi) [inline internal/handlers/sqlite/msg_explain.go:141]: Let's rename those response fields while we are on it

**Review by AlekSi — CHANGES_REQUESTED** [review]

**Review by rumyantseva — COMMENTED** [review]
I don't have my own questions about the PR (I agree with what Alexey asked).

File: website/docs/reference/glossary.md (hunk 5 above)
Comment (noisersup) [reply website/docs/reference/glossary.md:153]: Query pushdown is documentation page title. It's the general term for all types of pushdown. I don't understand what's wrong with it here, and what's the alternative?

File: integration/query_test.go (hunk 4 above)
Comment (noisersup) [reply integration/query_test.go]: fixed

File: website/docs/reference/glossary.md (hunk 5 above)
Comment (AlekSi) [reply website/docs/reference/glossary.md:153]: The alternative is "pushdown". It is always unclear, unlike "query pushdown" when is it sometimes unclear and sometimes means the same as "filter pushdown"

File: website/docs/reference/glossary.md (hunk 6 above)
Comment (noisersup) [reply website/docs/reference/glossary.md:149]: The issue mentioned glossary in TODO list, what changes in glossary should be applied?

File: website/docs/reference/glossary.md (hunk 5 above)
Comment (AlekSi) [reply website/docs/reference/glossary.md:153]: Let's focus on code changes for now, I will take a look on docs

Comment (noisersup) [reply website/docs/reference/glossary.md:153]: >  "query pushdown" when is it sometimes unclear and sometimes means the same as "filter pushdown"

I thought that was the issue we want to solve. This PR removes all "Query Pushdown" occurrences that mean "Filter Pushdown".

The standalone "Pushdown" seems a bit unclear to me, as it doesn't state what is "pushed down". That's Why Query Pushdown seems a better fit for general term, especially it's already used by some people (Googling `query pushdown` results with db related articles, and `pushdown` results with triceps exercise :dizzy_face:)

Comment (noisersup) [reply website/docs/reference/glossary.md:153]: > Let's focus on code changes for now, I will take a look on docs

Okay! I will quickly remove docs changes and re-request review

**Review by rumyantseva — DISMISSED** [review]
LGTM

**Review by rumyantseva — APPROVED** [review]

**Review by AlekSi — APPROVED** [review]

**Pull request merged**